test/cli.js

```diff
@@ -300,6 +300,43 @@ test('alex-cli', function (t) {
     }
   })
 
+  t.test('custom reporter', function (t) {
+    var fp = path.join('test', 'fixtures', 'profanity-sureness', 'two.md')
+
+    t.plan(1)
+
+    childProcess.exec('./cli.js --reporter json ' + fp, onexec)
+
+    const expectedJson = JSON.stringify([
+      {
+        path: fp,
+        cwd: process.cwd(),
+        history: [fp],
+        messages: []
+      }
+    ])
+
+    function onexec(err, stdout, stderr) {
+      t.deepEqual(
+        [err, stdout, stderr],
+        [null, '', expectedJson + '\n'],
+        'should work'
+      )
+    }
+  })
+
+  t.test("custom formatter that isn't installed", function (t) {
+    var fp = path.join('test', 'fixtures', 'profanity-sureness', 'two.md')
+
+    t.plan(1)
+
+    childProcess.exec('./cli.js --reporter doesntexist ' + fp, onexec)
+
+    function onexec(err, stdout, stderr) {
+      t.deepEqual([err, stderr, stdout], [null, '', ''], 'should work')
+    }
```

Comment:
All looks good, but, uhm, this is kinda weird 🤔 There should be some error, right?!

Here is a test in the underlying project that shows there is an error somewhere? https://github.com/unifiedjs/unified-engine/blob/39cbc12b53af119ea544b5d18984a39651139029/test/reporting.js#L285-L304

Reply:
fixed - didn't realise some errors go to stderr and others get passed to the callback